common/src/main/java/com/wynntils/screens/OverlayManagementScreen.java

```diff
@@ -0,0 +1,651 @@
+    private static final int ALIGNMENT_LINES_MAX_SECTIONS_PER_AXIS = 4;
+    private static final int ALIGNMENT_SNAP_DISTANCE = 1;
+    // Bigger this value is, the harder it is to not align overlay to alignment line
+    private static final double ALIGNMENT_SNAP_HARDNESS = 12;
+
+    private static final int BUTTON_WIDTH = 60;
+    private static final int BUTTON_HEIGHT = 20;
+    private static final int MAX_CLICK_DISTANCE = 10;
+
+    private static final List<Component> HELP_TOOLTIP_LINES = List.of(
+            new TextComponent("Resize the overlay by dragging the edges or corners."),
+            new TextComponent("Move it by dragging the center of the overlay."),
+            new TextComponent("Use your arrows to change vertical"),
+            new TextComponent("and horizontal alignment."),
+            new TextComponent("The overlay name will render respecting"),
+            new TextComponent("the current overlay alignments."));
+
+    private static final List<Component> CLOSE_TOOLTIP_LINES =
+            List.of(new TextComponent("Click here to stop editing and reset changes."));
+
+    private static final List<Component> TEST_TOOLTIP_LINES = List.of(
+            new TextComponent("Click here to toggle test mode."),
+            new TextComponent("In test mode, you can see how your overlay setup would look in-game,"),
+            new TextComponent("using preview render mode."));
+
+    private static final List<Component> APPLY_TOOLTIP_LINES =
+            List.of(new TextComponent("Click here to apply changes to current overlay."));
+
+    private static final Set<Float> verticalAlignmentLinePositions = new HashSet<>();
+    private static final Set<Float> horizontalAlignmentLinePositions = new HashSet<>();
+    private static final Map<Edge, Double> edgeAlignmentSnapMap = new HashMap<>();
+    private static final Map<Edge, Float> alignmentLinesToRender = new HashMap<>();
+
+    private static SelectionMode selectionMode = SelectionMode.None;
+    private static Overlay selectedOverlay = null;
+    private static Corner selectedCorner = null;
+    private static Edge selectedEdge = null;
+
+    private boolean testMode = false;
+
+    public OverlayManagementScreen(Overlay overlay) {
+        super(new TranslatableComponent("screens.wynntils.overlayManagement.name"));
+        selectedOverlay = overlay;
+    }
+
+    @Override
+    protected void init() {
+        setupButtons();
+        calculateAlignmentLinePositions();
+    }
+
+    @Override
+    public void onClose() {
+        reloadConfigForOverlay();
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        if (testMode) {
+            FontRenderer.getInstance()
+                    .renderText(
+                            poseStack,
+                            this.width / 2f,
+                            this.height - 160,
+                            new TextRenderTask(
```

Comment:
Breaking out the render task to a local variable might help spotless not f*ck this up completely...